In [1]:
#pip install --upgrade pip setuptools wheel

In [2]:
#pip install opencv-python

In [3]:
#pip install opencv-python==3.4.13.47

In [4]:
#pip install --upgrade pip setuptools wheel

In [5]:
#pip install cvzone

In [6]:
##pip install yolo

In [7]:
#from Ultralytics import YOLO

In [8]:
#pip install ultralytics-yolov5

In [9]:
#pip install ultralytics

In [10]:
#pip install ultralytics --proxy localhost:8892

In [2]:
from ultralytics import YOLO

In [9]:
#downloading weights n means nano
#gives smoother webcam performance
#YOU WILL GET ERROR, PUT 'yolov5n.pt' ONLY ON THE PATH
model=YOLO(r'C:\Users\fai-w\Desktop\weights\best.pt')
#yolov8l.pt -> webcam gets laggy

In [4]:
#to stop mage from closing when preprocessng is complete
import cv2

In [5]:
#!yolo task=detect mode=train model=yolov5n.pt data=c:\Users\fai-w\datasetFinal\data.yaml

In [11]:
#show results of image then closes image
results=model(r"C:\Users\fai-w\Desktop\download.jpg", show=True)
#keeps image open
cv2.waitKey(0)
cv2.destroyAllWindows()
results


image 1/1 C:\Users\fai-w\Desktop\download.jpg: 640x640 1 Jacket, 154.1ms
Speed: 6.0ms preprocess, 154.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)


[ultralytics.yolo.engine.results.Results object with attributes:
 
 boxes: ultralytics.yolo.engine.results.Boxes object
 keypoints: None
 keys: ['boxes']
 masks: None
 names: {0: 'Hoodie', 1: 'Jacket', 2: 'Mid-lenght dress', 3: 'Pants', 4: 'Tshirt', 5: 'dress', 6: 'skirt'}
 orig_img: array([[[253, 253, 253],
         [253, 253, 253],
         [253, 253, 253],
         ...,
         [253, 253, 253],
         [253, 253, 253],
         [253, 253, 253]],
 
        [[253, 253, 253],
         [253, 253, 253],
         [253, 253, 253],
         ...,
         [253, 253, 253],
         [253, 253, 253],
         [253, 253, 253]],
 
        [[253, 253, 253],
         [253, 253, 253],
         [253, 253, 253],
         ...,
         [253, 253, 253],
         [253, 253, 253],
         [253, 253, 253]],
 
        ...,
 
        [[253, 253, 253],
         [253, 253, 253],
         [253, 253, 253],
         ...,
         [253, 253, 253],
         [253, 253, 253],
         [253, 253, 253]],
 
        [

In [16]:
#for r in results:
#        boxes=r.boxes
#        #get Xs and Ys of each box
#        for box in boxes:
#            x1, y1, x2, y2 = box.xyxy[0]
#            print(x1, y1, x2, y2)
#IDEA USE BETTER ACCURACY WHEN PEOPLE WANT TO UPLOAD AN IMAGE 
#BUT IN REALTIME OBJECT DETECTION GET WORSE WEIGHT BUT ADJUST MANUALLY

In [17]:
import pandas as pd
import numpy as np

#colorss=pd.read_csv('D:\color_names.csv')

In [44]:
#pip install streamlit

     ---------------------------------------- 0.0/8.9 MB ? eta -:--:--
     ---------------------------------------- 0.0/8.9 MB 640.0 kB/s eta 0:00:14
     ---------------------------------------- 0.0/8.9 MB 393.8 kB/s eta 0:00:23
     ---------------------------------------- 0.1/8.9 MB 651.6 kB/s eta 0:00:14
      --------------------------------------- 0.1/8.9 MB 726.2 kB/s eta 0:00:13
      --------------------------------------- 0.2/8.9 MB 756.6 kB/s eta 0:00:12
     - -------------------------------------- 0.3/8.9 MB 923.9 kB/s eta 0:00:10
     - -------------------------------------- 0.3/8.9 MB 896.4 kB/s eta 0:00:10
     - -------------------------------------- 0.4/8.9 MB 967.1 kB/s eta 0:00:09
     - -------------------------------------- 0.4/8.9 MB 981.2 kB/s eta 0:00:09
     -- ------------------------------------- 0.5/8.9 MB 938.8 kB/s eta 0:00:09
     -- ------------------------------------- 0.5/8.9 MB 1.0 MB/s eta 0:00:09
     -- ------------------------------------- 0.6/8

In [7]:
import streamlit as st
 
# Title
st.title("Hello GeeksForGeeks !!!")

2023-06-20 19:48:59.688 
  command:

    streamlit run D:\anacnda3\envs\cudaenv\lib\site-packages\ipykernel_launcher.py [ARGUMENTS]


DeltaGenerator()

In [18]:
#colorss.info()

In [12]:
#####TESTINGGGG#############
#importing cvzone to show detections
import cvzone
import numpy as np
#webcam object
cam=cv2.VideoCapture(0)
#width
cam.set(3, 1280)
#height
cam.set(4, 720)

#LOWER AND UPPER RANGE OF MULTIPLE COLORS

lower_black = np.array([0,0,0])
upper_black = np.array([250,255,30])

lower_white = np.array([0,0,255])
upper_white = np.array([0,0,255])

lower_red = np.array([0,150,50])
upper_red = np.array([10,255,255])

lower_green = np.array([45,150,50])
upper_green = np.array([65,255,255])

lower_yellow = np.array([25,150,50])
upper_yellow = np.array([35,255,255])

lower_light_blue = np.array([95,150,0])
upper_light_blue = np.array([110,255,255])

lower_orange = np.array([15,150,0])
upper_orange = np.array([25,255,255])

lower_dark_pink = np.array([160,150,0])
upper_dark_pink = np.array([170,255,255])

lower_pink = np.array([145,150,0])
upper_pink = np.array([155,255,255])

lower_cyan = np.array([85,150,0])
upper_cyan = np.array([95,255,255])

lower_dark_blue = np.array([115,150,0])
upper_dark_blue = np.array([125,255,255])

#purple
lower_purple = np.array([129, 50, 70])
#np.array([129, 50, 70])138,43,226
upper_purple = np.array([158, 255, 255])
#np.array([158, 255, 255])255,0,255
#light purple
#upper_light_purple = np.array([220,208,255])
#lower_light_purple = np.array([181,126,220])
#dark purple
#upper_dark_purple = np.array([145,92,131])
#lower_dark_purple = np.array([50,23,77])


#turqise
lower_turq = np.array([0,206,209])
upper_turq = np.array([175,238,238])



#TO KEEP THE WEBCAM RUNNING
while True:
    #IF CAM IS FOUND, SAVE IT TO "img"
    success, img = cam.read()
    #YOLO MODEL RESULTS
    results=model(img, stream=True)
    #CAPTURE COLORS AND TURN THEM TO "HSV" VALUES
    imgC = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
    #range of colors
    yellow = cv2.inRange(imgC, lower_yellow, upper_yellow)
    red = cv2.inRange(imgC, lower_red, upper_red)
    dblue = cv2.inRange(imgC, lower_dark_blue, upper_dark_blue)
    green = cv2.inRange(imgC, lower_green, upper_green)
    pink = cv2.inRange(imgC, lower_pink, upper_pink)
    lblue = cv2.inRange(imgC, lower_light_blue, upper_light_blue)
    purple = cv2.inRange(imgC, lower_purple, upper_purple)
    orange = cv2.inRange(imgC, lower_orange, upper_orange)
    cyan = cv2.inRange(imgC, lower_cyan, upper_cyan)
    turq = cv2.inRange(imgC, lower_turq, upper_turq)
    
    #lpurple = cv2.inRange(imgC, lower_light_purple, upper_light_purple)
    #dpurple = cv2.inRange(imgC, lower_dark_purple, upper_dark_purple)
    #morphological transformation and dilation
    #kernal=np.ones((5,5), 'uint8')
    #red=cv2.dilate(red, kernal)
    #resr=cv2.bitwise_and(img, img, mask = red)
    #blue=cv2.dilate(blue, kernal)
    #resb=cv2.bitwise_and(img, img, mask = blue)
    #yellow=cv2.dilate(yellow, kernal)
    #resy=cv2.bitwise_and(img, img, mask = red)
    
    #tracking colors
    (contours, hierarchy)=cv2.findContours(red, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'red', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    
    #BLUE
    (contours, hierarchy)=cv2.findContours(dblue, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'dark blue', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #YELLOW
    (contours, hierarchy)=cv2.findContours(yellow, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
            
    #green
    (contours, hierarchy)=cv2.findContours(green, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'green', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #pink
    (contours, hierarchy)=cv2.findContours(pink, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'pink', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #lblue
    (contours, hierarchy)=cv2.findContours(lblue, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'light blue', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    
    #purple
    (contours, hierarchy)=cv2.findContours(purple, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'purple', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #ORANGE
    (contours, hierarchy)=cv2.findContours(orange, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'orange', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #CYAN
    (contours, hierarchy)=cv2.findContours(cyan, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'cyan', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
            
    #TURQ
    (contours, hierarchy)=cv2.findContours(turq, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'turq', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #cv2.imshow('mask', mask)
    #cv2.imshow('mask', mask)
    
    #cv2.imshow('mask', mask)
    #contour, hierarchy = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    #if len(contour)!=0:
    #    for cont in contour:
    #        if cv2.contourArea(cont)>1000:
    #            x, y, w, h=cv2.boundingRect(cont)
    #            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 3)
    #            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (251,255,255), 2)
    #cam.set(CV_CAP_PROP_BUFFERSIZE, 3)
    #results.boxes
    #IF CAN NOT THERE
    if not cam.isOpened():
        print('Unable to load camera.')
        sleep(5)
        pass
    #FOR CAPTURING BOUNDRIES OF THE YOLO PREDICTIONS 
    # Capture frame-by-frame
    ret, frame = cam.read()
    #end of extra
    #loop through every frame
    for r in results:
        boxes=r.boxes
        #get Xs and Ys of each box
        for box in boxes:
            x1, y1, x2, y2 = box.xyxy[0]
            print(x1, y1, x2, y2)
    #cv2.imshow("image", img)
    #TO QUIT WEBCAM, PRESS "q" (SOMETIMES YOU HAVE TO HOLD IT AND OTHER TIMES YOU CAN JUST PRESS IT)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cam.release()
cv2.destroyAllWindows()



0: 384x640 1 Jacket, 126.7ms
Speed: 3.0ms preprocess, 126.7ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(413.1975) tensor(382.2365) tensor(803.2441) tensor(714.9299)


0: 384x640 1 Hoodie, 1 Jacket, 144.6ms
Speed: 3.0ms preprocess, 144.6ms inference, 1.1ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 103.4ms


tensor(401.2338) tensor(389.4643) tensor(995.2722) tensor(716.3684)
tensor(966.0951) tensor(203.1566) tensor(1280.) tensor(717.4777)
tensor(404.3114) tensor(401.9263) tensor(1225.8459) tensor(716.1241)


Speed: 4.0ms preprocess, 103.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Jacket, 102.3ms
Speed: 2.0ms preprocess, 102.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 102.0ms


tensor(404.4465) tensor(392.6226) tensor(792.4751) tensor(715.2907)
tensor(1057.8759) tensor(108.6626) tensor(1280.) tensor(720.)
tensor(399.7965) tensor(394.5255) tensor(1239.4017) tensor(716.6122)


Speed: 3.0ms preprocess, 102.0ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 112.4ms
Speed: 2.0ms preprocess, 112.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 120.2ms


tensor(1028.8911) tensor(124.2614) tensor(1279.9961) tensor(720.)
tensor(404.4075) tensor(401.9509) tensor(1023.6729) tensor(715.1521)


Speed: 2.0ms preprocess, 120.2ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 113.4ms
Speed: 3.0ms preprocess, 113.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Jacket, 103.1ms
Speed: 2.0ms preprocess, 103.1ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 94.0ms
Speed: 2.0ms preprocess, 94.0ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)


tensor(943.2618) tensor(189.9346) tensor(1279.8778) tensor(720.)
tensor(405.0208) tensor(407.2775) tensor(798.2151) tensor(715.2215)



0: 384x640 1 Hoodie, 104.5ms
Speed: 2.1ms preprocess, 104.5ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 104.1ms


tensor(946.2512) tensor(189.0446) tensor(1280.) tensor(717.8805)


Speed: 3.0ms preprocess, 104.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 103.2ms
Speed: 3.0ms preprocess, 103.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 117.5ms


tensor(397.0951) tensor(380.4675) tensor(807.9601) tensor(715.4185)


Speed: 3.0ms preprocess, 117.5ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 97.4ms
Speed: 3.0ms preprocess, 97.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 97.2ms
Speed: 4.0ms preprocess, 97.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 109.3ms


tensor(364.2381) tensor(351.2200) tensor(773.9748) tensor(715.8288)


Speed: 3.0ms preprocess, 109.3ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 96.0ms
Speed: 2.0ms preprocess, 96.0ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 97.3ms
Speed: 2.0ms preprocess, 97.3ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(937.0224) tensor(183.4775) tensor(1280.) tensor(719.1065)


0: 384x640 (no detections), 124.1ms
Speed: 2.0ms preprocess, 124.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Tshirt, 121.4ms
Speed: 3.0ms preprocess, 121.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 115.4ms


tensor(349.2711) tensor(346.1292) tensor(887.2870) tensor(715.0498)
tensor(350.9216) tensor(345.8700) tensor(888.2112) tensor(715.0194)


Speed: 2.0ms preprocess, 115.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 116.4ms
Speed: 2.0ms preprocess, 116.4ms inference, 1.1ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 115.3ms
Speed: 2.0ms preprocess, 115.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Tshirt, 111.1ms
Speed: 2.0ms preprocess, 111.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(349.4706) tensor(345.0076) tensor(836.5552) tensor(714.5609)
tensor(351.1699) tensor(347.1263) tensor(835.0015) tensor(713.9668)


0: 384x640 1 Tshirt, 116.3ms
Speed: 3.0ms preprocess, 116.3ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(339.3465) tensor(337.3592) tensor(864.2590) tensor(715.4387)
tensor(344.1704) tensor(349.9928) tensor(877.4807) tensor(714.0736)
tensor(344.6579) tensor(349.5470) tensor(866.1935) tensor(715.1500)


0: 384x640 1 Hoodie, 1 Tshirt, 118.7ms
Speed: 3.0ms preprocess, 118.7ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 105.7ms
Speed: 3.0ms preprocess, 105.7ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(342.8479) tensor(349.2294) tensor(835.6108) tensor(713.5394)


0: 384x640 (no detections), 116.1ms
Speed: 2.0ms preprocess, 116.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 135.2ms
Speed: 4.0ms preprocess, 135.2ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 105.2ms
Speed: 2.0ms preprocess, 105.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(352.2018) tensor(355.2684) tensor(882.0643) tensor(713.5587)


0: 384x640 1 Hoodie, 1 Tshirt, 133.1ms
Speed: 3.0ms preprocess, 133.1ms inference, 1.1ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 110.0ms


tensor(351.9563) tensor(353.3406) tensor(848.9163) tensor(714.0395)
tensor(350.8687) tensor(352.6295) tensor(849.1170) tensor(714.7748)


Speed: 2.0ms preprocess, 110.0ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 118.3ms
Speed: 4.0ms preprocess, 118.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 121.5ms


tensor(348.2112) tensor(357.3204) tensor(837.2585) tensor(713.9374)


Speed: 2.0ms preprocess, 121.5ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 114.8ms
Speed: 4.0ms preprocess, 114.8ms inference, 0.9ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 112.9ms
Speed: 3.0ms preprocess, 112.9ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)



tensor(340.5630) tensor(356.5612) tensor(892.3629) tensor(714.4898)


0: 384x640 1 Jacket, 117.1ms
Speed: 3.0ms preprocess, 117.1ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 123.1ms


tensor(342.7401) tensor(356.0703) tensor(854.9539) tensor(716.2477)
tensor(349.7342) tensor(358.8361) tensor(833.6703) tensor(713.0936)


Speed: 2.0ms preprocess, 123.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 123.0ms
Speed: 3.0ms preprocess, 123.0ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 109.1ms
Speed: 3.0ms preprocess, 109.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)


tensor(337.0312) tensor(338.6679) tensor(859.8702) tensor(714.8116)



0: 384x640 (no detections), 121.2ms
Speed: 4.0ms preprocess, 121.2ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 116.9ms
Speed: 3.0ms preprocess, 116.9ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Jacket, 113.0ms
Speed: 3.0ms preprocess, 113.0ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 97.8ms
Speed: 3.0ms preprocess, 97.8ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)


tensor(344.9892) tensor(354.4766) tensor(849.1766) tensor(713.6923)
tensor(949.8379) tensor(209.7207) tensor(1280.) tensor(717.8931)



0: 384x640 1 Hoodie, 105.7ms
Speed: 3.0ms preprocess, 105.7ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 1 Jacket, 98.2ms


tensor(965.9885) tensor(167.3766) tensor(1279.7415) tensor(720.)
tensor(336.1833) tensor(361.6015) tensor(824.1973) tensor(715.3546)
tensor(961.7828) tensor(185.4453) tensor(1279.8409) tensor(720.)


Speed: 3.0ms preprocess, 98.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 2 Hoodies, 127.3ms
Speed: 2.0ms preprocess, 127.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 99.5ms


tensor(343.8135) tensor(364.4341) tensor(862.5084) tensor(714.1862)
tensor(954.1324) tensor(184.8333) tensor(1280.) tensor(718.5635)
tensor(331.8219) tensor(355.8573) tensor(943.7850) tensor(715.1687)


Speed: 4.0ms preprocess, 99.5ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 105.3ms
Speed: 2.0ms preprocess, 105.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 104.2ms
Speed: 4.0ms preprocess, 104.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 97.4ms
Speed: 2.1ms preprocess, 97.4ms inference, 1.1ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Tshirt, 97.2ms
Speed: 3.0ms preprocess, 97.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 97.1ms


tensor(314.1831) tensor(350.4034) tensor(981.2551) tensor(715.2732)


Speed: 3.0ms preprocess, 97.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 112.4ms
Speed: 2.0ms preprocess, 112.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 109.3ms
Speed: 3.1ms preprocess, 109.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 101.3ms
Speed: 3.0ms preprocess, 101.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 128.7ms
Speed: 3.0ms preprocess, 128.7ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 108.5ms
Speed: 4.0ms preprocess, 108.5ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 2 Hoodies, 1 Jacket, 95.1ms


tensor(349.6423) tensor(366.1492) tensor(1089.3542) tensor(713.4287)
tensor(357.4632) tensor(362.1148) tensor(895.3390) tensor(713.5657)
tensor(945.3494) tensor(185.8237) tensor(1280.) tensor(718.4428)
tensor(1076.0990) tensor(156.6309) tensor(1279.9957) tensor(720.)


Speed: 3.0ms preprocess, 95.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 110.2ms
Speed: 3.1ms preprocess, 110.2ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 108.6ms
Speed: 2.0ms preprocess, 108.6ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 106.1ms
Speed: 3.1ms preprocess, 106.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 3 Hoodies, 96.4ms


tensor(357.8581) tensor(365.7866) tensor(843.6382) tensor(713.4063)
tensor(358.2945) tensor(370.3957) tensor(846.9553) tensor(713.3107)
tensor(932.1776) tensor(188.1539) tensor(1280.) tensor(718.1423)
tensor(1056.8531) tensor(177.7568) tensor(1280.) tensor(720.)


Speed: 2.0ms preprocess, 96.4ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 3 Hoodies, 1 Jacket, 100.3ms
Speed: 2.1ms preprocess, 100.3ms inference, 2.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 90.1ms


tensor(937.9224) tensor(178.1392) tensor(1280.) tensor(718.0758)
tensor(362.9749) tensor(376.6330) tensor(839.6980) tensor(713.1552)
tensor(1098.3379) tensor(154.1040) tensor(1279.9458) tensor(720.)
tensor(362.5595) tensor(376.6778) tensor(844.6827) tensor(713.1160)
tensor(360.5646) tensor(366.4030) tensor(856.3649) tensor(713.5144)


Speed: 2.0ms preprocess, 90.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Hoodie, 107.4ms
Speed: 2.0ms preprocess, 107.4ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 1 Jacket, 100.3ms


tensor(358.5412) tensor(366.8686) tensor(819.5534) tensor(713.3131)
tensor(349.3428) tensor(366.6837) tensor(860.2214) tensor(714.8075)


Speed: 2.0ms preprocess, 100.3ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 118.3ms
Speed: 2.0ms preprocess, 118.3ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)


In [8]:
#importing cvzone to show detections
import cvzone
import numpy as np
#webcam object
cam=cv2.VideoCapture(0)
#width
cam.set(3, 1280)
#height
cam.set(4, 720)

#LOWER AND UPPER RANGE OF MULTIPLE COLORS

lower_black = np.array([0,0,0])
upper_black = np.array([250,255,30])

lower_white = np.array([0,0,255])
upper_white = np.array([0,0,255])

lower_red = np.array([0,150,50])
upper_red = np.array([10,255,255])

lower_green = np.array([45,150,50])
upper_green = np.array([65,255,255])

lower_yellow = np.array([25,150,50])
upper_yellow = np.array([35,255,255])

lower_light_blue = np.array([95,150,0])
upper_light_blue = np.array([110,255,255])

lower_orange = np.array([15,150,0])
upper_orange = np.array([25,255,255])

lower_dark_pink = np.array([160,150,0])
upper_dark_pink = np.array([170,255,255])

lower_pink = np.array([145,150,0])
upper_pink = np.array([155,255,255])

lower_cyan = np.array([85,150,0])
upper_cyan = np.array([95,255,255])

lower_dark_blue = np.array([115,150,0])
upper_dark_blue = np.array([125,255,255])

#purple
lower_purple = np.array([129, 50, 70])
#np.array([129, 50, 70])138,43,226
upper_purple = np.array([158, 255, 255])
#np.array([158, 255, 255])255,0,255
#light purple
#upper_light_purple = np.array([220,208,255])
#lower_light_purple = np.array([181,126,220])
#dark purple
#upper_dark_purple = np.array([145,92,131])
#lower_dark_purple = np.array([50,23,77])


#turqise
lower_turq = np.array([0,206,209])
upper_turq = np.array([175,238,238])



#TO KEEP THE WEBCAM RUNNING
while True:
    #IF CAM IS FOUND, SAVE IT TO "img"
    success, img = cam.read()
    #YOLO MODEL RESULTS
    results=model(img, stream=True)
    #CAPTURE COLORS AND TURN THEM TO "HSV" VALUES
    imgC = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
    #range of colors
    yellow = cv2.inRange(imgC, lower_yellow, upper_yellow)
    red = cv2.inRange(imgC, lower_red, upper_red)
    dblue = cv2.inRange(imgC, lower_dark_blue, upper_dark_blue)
    green = cv2.inRange(imgC, lower_green, upper_green)
    pink = cv2.inRange(imgC, lower_pink, upper_pink)
    lblue = cv2.inRange(imgC, lower_light_blue, upper_light_blue)
    purple = cv2.inRange(imgC, lower_purple, upper_purple)
    orange = cv2.inRange(imgC, lower_orange, upper_orange)
    cyan = cv2.inRange(imgC, lower_cyan, upper_cyan)
    turq = cv2.inRange(imgC, lower_turq, upper_turq)

    #lpurple = cv2.inRange(imgC, lower_light_purple, upper_light_purple)
    #dpurple = cv2.inRange(imgC, lower_dark_purple, upper_dark_purple)
    #morphological transformation and dilation
    #kernal=np.ones((5,5), 'uint8')
    #red=cv2.dilate(red, kernal)
    #resr=cv2.bitwise_and(img, img, mask = red)
    #blue=cv2.dilate(blue, kernal)
    #resb=cv2.bitwise_and(img, img, mask = blue)
    #yellow=cv2.dilate(yellow, kernal)
    #resy=cv2.bitwise_and(img, img, mask = red)
    
    #tracking colors
    (contours, hierarchy)=cv2.findContours(red, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'red', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    
    #BLUE
    (contours, hierarchy)=cv2.findContours(dblue, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'dark blue', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #YELLOW
    (contours, hierarchy)=cv2.findContours(yellow, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
            
    #green
    (contours, hierarchy)=cv2.findContours(green, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'green', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #pink
    (contours, hierarchy)=cv2.findContours(pink, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'pink', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #lblue
    (contours, hierarchy)=cv2.findContours(lblue, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'light blue', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    
    #purple
    (contours, hierarchy)=cv2.findContours(purple, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'purple', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #ORANGE
   
    #CYAN
    (contours, hierarchy)=cv2.findContours(cyan, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'cyan', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
            
    #TURQ
    (contours, hierarchy)=cv2.findContours(turq, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'turq', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #cv2.imshow('mask', mask)
    #cv2.imshow('mask', mask)
    
    #cv2.imshow('mask', mask)
    #contour, hierarchy = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    #if len(contour)!=0:
    #    for cont in contour:
    #        if cv2.contourArea(cont)>1000:
    #            x, y, w, h=cv2.boundingRect(cont)
    #            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 3)
    #            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (251,255,255), 2)
    #cam.set(CV_CAP_PROP_BUFFERSIZE, 3)
    #results.boxes
    #IF CAN NOT THERE
    if not cam.isOpened():
        print('Unable to load camera.')
        sleep(5)
        pass
    #FOR CAPTURING BOUNDRIES OF THE YOLO PREDICTIONS 
    # Capture frame-by-frame
    ret, frame = cam.read()
    #end of extra
    #loop through every frame
    for r in results:
        boxes=r.boxes
        #get Xs and Ys of each box
        for box in boxes:
            x1, y1, x2, y2 = box.xyxy[0]
            print(x1, y1, x2, y2)
            (contours, hierarchy)=cv2.findContours(orange, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
            for pic, conn in enumerate(contours):
                area=cv2.contourArea(conn)
                #x1. try allign to
                if(area>500):
                    x, y, w, h = cv2.boundingRect(conn)
                #if(x1<x and y1<y and x2>w and y2>h):
                    cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
                    cv2.putText(img, 'orange', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
                
    cv2.imshow("image", img)
    #TO QUIT WEBCAM, PRESS "q" (SOMETIMES YOU HAVE TO HOLD IT AND OTHER TIMES YOU CAN JUST PRESS IT)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cam.release()
cv2.destroyAllWindows()



0: 384x640 (no detections), 145.3ms
Speed: 7.9ms preprocess, 145.3ms inference, 3.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 126.4ms
Speed: 1.7ms preprocess, 126.4ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 122.9ms
Speed: 2.0ms preprocess, 122.9ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 113.1ms
Speed: 3.0ms preprocess, 113.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 127.9ms
Speed: 3.0ms preprocess, 127.9ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 119.5ms
Speed: 4.0ms preprocess, 119.5ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 102.1ms
Speed: 2.0ms preprocess, 102.1ms inference, 0.0ms postprocess per image at shape (1, 3, 640, 640)

0: 384x640 (no detections), 102.7ms
Speed: 2.0ms prepr

In [ ]:
# حولت الأكواد القديمة لمارك أب احتياط 

#####TESTINGGGG#############
#importing cvzone to show detections
import cvzone
import numpy as np
#webcam object
cam=cv2.VideoCapture(0)
#width
cam.set(3, 1280)
#height
cam.set(4, 720)

####colors#####

#YELLOW
#lower range of yellow
#yellow_lower = np.array([15, 150, 20])
#upper range of yellow
#yellow_upper = np.array([35, 255, 255])
yellow_lower = np.array([15, 150, 20])
#upper range of yellow
yellow_upper = np.array([35, 255, 255])
#BLUE
#lower range of blue
#blue_lower = np.array([99, 115, 150])
#upper range of blue
#blue_upper = np.array([60, 255, 255])
blue_lower = np.array([0, 25, 125])
#upper range of blue
blue_upper = np.array([183, 220, 255])
#RED
#lower range of red
#red_lower = np.array([136, 87, 111])
#upper range of red
#red_upper = np.array([180, 255, 255])
red_lower = np.array([124, 0, 0])
#upper range of red
red_upper = np.array([255, 67, 67])
#PINK
pink_lower = np.array([164, 43, 43])
#upper range of red
pink_upper = np.array([255, 199, 199])
#PURPLE
#GREEN
green_lower = np.array([0, 48, 20])
#upper range of yellow
green_upper = np.array([210, 255, 226])


while True:
    success, img = cam.read()
    results=model(img, stream=True)
    imgC = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
    #range of colors
    yellow = cv2.inRange(imgC, yellow_lower, yellow_upper)
    red = cv2.inRange(imgC, red_lower, red_upper)
    blue = cv2.inRange(imgC, blue_lower, blue_upper)
    #morphological transformation and dilation
    #kernal=np.ones((5,5), 'uint8')
    #red=cv2.dilate(red, kernal)
    #resr=cv2.bitwise_and(img, img, mask = red)
    #blue=cv2.dilate(blue, kernal)
    #resb=cv2.bitwise_and(img, img, mask = blue)
    #yellow=cv2.dilate(yellow, kernal)
    #resy=cv2.bitwise_and(img, img, mask = red)
    
    #tracking colors
    (contours, hierarchy)=cv2.findContours(red, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'red', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #BLUE
    (contours, hierarchy)=cv2.findContours(blue, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'blue', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
    #YELLOW
    (contours, hierarchy)=cv2.findContours(yellow, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    for pic, conn in enumerate(contours):
        area=cv2.contourArea(conn)
        if(area>500):
            x, y, w, h = cv2.boundingRect(conn)
            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 2)
            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (251,255,255), 2)
            
    
    #cv2.imshow('mask', mask)
    #contour, hierarchy = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    #if len(contour)!=0:
    #    for cont in contour:
    #        if cv2.contourArea(cont)>1000:
    #            x, y, w, h=cv2.boundingRect(cont)
    #            cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 3)
    #            cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (251,255,255), 2)
    #cam.set(CV_CAP_PROP_BUFFERSIZE, 3)
    #results.boxes
    #extra
    if not cam.isOpened():
        print('Unable to load camera.')
        sleep(5)
        pass

    # Capture frame-by-frame
    ret, frame = cam.read()
    #end of extra
    #loop through every frame
    for r in results:
        boxes=r.boxes
        #get Xs and Ys of each box
        for box in boxes:
            x1, y1, x2, y2 = box.xyxy[0]
            print(x1, y1, x2, y2)
    #cv2.imshow("image", img)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cam.release()
cv2.destroyAllWindows()
    #if cv2.getWindowProperty(img, cv2.WND_PROP_VISIBLE) <1:
        #break
#cv2.destroyAllWindows()

#importing cvzone to show detections
import cvzone
import numpy as np
#webcam object
cam=cv2.VideoCapture(0)
#width
cam.set(3, 1280)
#height
cam.set(4, 720)
#lower range of color
lower = np.array([15, 150, 20])
#upper range of color
upper = np.array([35, 255, 255])
while True:
    success, img = cam.read()
    results=model(img, stream=True)
    imgC = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
    mask = cv2.inRange(imgC, lower, upper)
    cv2.imshow('mask', mask)
    contour, hierarchy = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    if len(contour)!=0:
        for cont in contour:
            if cv2.contourArea(cont)>1000:
                x, y, w, h=cv2.boundingRect(cont)
                cv2.rectangle(img, (x, y), (x+w,y+h), (251, 255, 255), 3)
                cv2.putText(img, 'yellow', (x, y+h+10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (251,255,255), 2)
    #cam.set(CV_CAP_PROP_BUFFERSIZE, 3)
    #results.boxes
    #extra
    if not cam.isOpened():
        print('Unable to load camera.')
        sleep(5)
        pass

    # Capture frame-by-frame
    ret, frame = cam.read()
    #end of extra
    #loop through every frame
    for r in results:
        boxes=r.boxes
        #get Xs and Ys of each box
        for box in boxes:
            x1, y1, x2, y2 = box.xyxy[0]
            print(x1, y1, x2, y2)
    #cv2.imshow("image", img)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break
cam.release()
cv2.destroyAllWindows()
    #if cv2.getWindowProperty(img, cv2.WND_PROP_VISIBLE) <1:
        #break
#cv2.destroyAllWindows()

In [36]:
x1.item()

320.8778076171875

In [ ]:
#pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118

In [ ]:
#import torch
#import torchvision.models as models

In [ ]:
#torch.cuda.is_available()

In [ ]:
#torch._C.has_cuda

In [ ]:
#pip install tensorflow-gpu

In [ ]:
#!nvidia-smi